frontend/src/queries/nodes/DataVisualization/DataVisualization.tsx

```diff
@@ -0,0 +1,81 @@
+import { LemonDivider } from '@posthog/lemon-ui'
+import { BindLogic, useValues } from 'kea'
+import { uuid } from 'lib/utils'
+import { useCallback, useState } from 'react'
+
+import { AnyResponseType, DataVisualizationNode, HogQLQuery, NodeKind } from '~/queries/schema'
+import { QueryContext } from '~/queries/types'
+import { ChartDisplayType } from '~/types'
+
+import { DataTable } from '../DataTable/DataTable'
+import { HogQLQueryEditor } from '../HogQLQuery/HogQLQueryEditor'
+import { Chart } from './Components/Chart'
+import { ElapsedTime } from './Components/ElapsedTime'
+import { Reload } from './Components/Reload'
+import { TableDisplay } from './Components/TableDisplay'
+import { dataVisualizationLogic } from './dataVisualizationLogic'
+
+interface DataTableVisualizationProps {
+    uniqueKey?: string | number
+    query: DataVisualizationNode
+    setQuery?: (query: DataVisualizationNode) => void
+    context?: QueryContext
+    /* Cached Results are provided when shared or exported,
+    the data node logic becomes read only implicitly */
+    cachedResults?: AnyResponseType
+}
+
+export function DataTableVisualization(props: DataTableVisualizationProps): JSX.Element {
+    const [key] = useState(props.uniqueKey?.toString() ?? uuid())
```

Comment:
If there's no `uniqueKey` in props, the `uuid` will be different on every render, likely making a mess.